### PR TITLE
feat: Detect override of control "rerender"

### PR DESCRIPTION
**Pull request by matz3**
JIRA: CPOUI5FOUNDATION-939

### CODE REVIEW

File: src/linter/ui5Types/SourceFileLinter.ts [hunk 1]
```diff
@@ -539,6 +540,19 @@ export default class SourceFileLinter {
 		}
 	}
 
+	analyzeControlRerenderMethod(node: ts.ClassDeclaration) {
+		const className = node.name?.getText() ?? "<unknown>";
+		const rerenderMethod = node.members.find((member) => {
+			return ts.isMethodDeclaration(member) &&
```
Comment (d3xter666) [inline src/linter/ui5Types/SourceFileLinter.ts:546]: Do we support `ArrowExpression` there?

File: test/fixtures/linter/rules/renderer/ControlRerenderOverride.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+sap.ui.define(["sap/ui/core/Control"], function(Control, Button) {
+	const Example1 = Control.extend("sap.ui.demo.linter.controls.Example1", {
+		metadata: {},
+
+		rerender: function() {
```
Comment (d3xter666) [inline test/fixtures/linter/rules/renderer/ControlRerenderOverride.js:5]: What about destructuring or assignment?

```
	const Example1 = Control.extend("sap.ui.demo.linter.controls.Example1", {
		metadata: {},

		rerender: function() {
Comment
What about destructuring or assignment?

```
      var rerender = function() {
			console.log("Overriding rerender method");
			return Control.prototype.rerender.apply(this, arguments);
		},
	const Example1 = Control.extend("sap.ui.demo.linter.controls.Example1", {
		metadata: {},

		rerender, // or rerender: rerender
....
```

File: src/linter/ui5Types/SourceFileLinter.ts (hunk 1 above)
Comment (matz3) [reply src/linter/ui5Types/SourceFileLinter.ts:546]: Not yet, good catch!

File: test/fixtures/linter/rules/renderer/ControlRerenderOverride.js (hunk 2 above)
Comment (matz3) [reply test/fixtures/linter/rules/renderer/ControlRerenderOverride.js:5]: I added support for a simple assignment as shown above. I do not think that we need to handle more complex scenarios.

**Review by d3xter666 — APPROVED** [review]

**Review by RandomByte — COMMENTED** [review]
LGTM

**Pull request merged**